python-avd/pyavd/_eos_designs/shared_utils/routing.py

```diff
@@ -36,6 +36,9 @@ def overlay_address_families(self: SharedUtils) -> list[str]:
         if self.overlay_routing_protocol in ["ebgp", "ibgp"]:
             default_overlay_address_families = self.node_type_key_data.default_overlay_address_families
             return self.node_config.overlay_address_families._as_list() or default_overlay_address_families._as_list()
+        # TODO: discuss if this is the best place
```

Comment:
You need to evaluate overlay_routing_protocol with the toggle. So return "none" for WAN routers unless the toggle is set (for which we will honor the setting).

On this setting I would add `or self.is_wan_router` on line 36 so we just honor the setting for WAN routers. I suppose the default is `['evpn']` on those.